Fix:
error: implicitly declaring library function 'memcpy' with type 'void *(void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
--- readpng.c.orig	2005-10-23 14:36:29.000000000 -0500
+++ readpng.c	2022-01-08 14:14:46.000000000 -0600
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <png.h>
 
 #include <X11/Xlib.h>
